app/controllers/tasks_controller.rb

```diff
@@ -1,12 +1,17 @@
 # frozen_string_literal: true
 
 class TasksController < ApplicationController
+  before_action :load_list
   before_action :set_task, only: %i[show edit update destroy]
 
   # GET /tasks
   # GET /tasks.json
   def index
-    @tasks = Task.all
+    @tasks = if params.fetch(:filter, nil).present?
+               current_user.tasks
+             else
+               current_user.tasks.pending
+             end
```

Comment:
Given the task resources are nested under the list routes, is there any problem with loading tasks like this?

Reply:
Woops. That's a genuine derp right there. You'd want to ensure that it's scoping the tasks returned to be scoped to the `current_user` as well as the `@list.id`. Really what I'd do is like `@tasks = current_user.tasks.with_list(@list)` and then chain on the remaining scopes (in the case of pending)

with_list would just be `:scope, :with_list -> (list) { where(list_id: list.id) }`